backend/test/customerTests/controller.test.ts

```diff
@@ -0,0 +1,181 @@
+        expect(res.status).to.equal(400);
+        app.shutdown();
+    });
+
+    it('Test find customer by ID route', async () => {
+        const mockCustomerService = sandbox.createStubInstance(CustomerService);
+        mockCustomerService.findCustomerById.resolves('foo');
+        const app = new App({
+            port: testPort,
+            controllers: [new CustomerController(mockCustomerService)],
+            middleWares: [bodyParser.json(), bodyParser.urlencoded({ extended: true })]
+        });
+        app.listen();
+        const res = await request(app.app)
+            .get('/customer/1')
+            .set('Accept', 'application/json')
+            .set('Authorization', 'bearer ' + token);
+        expect(res.body).to.equal('foo');
+        expect(res.status).to.equal(200);
+        app.shutdown();
+    });
+
+    it('Test update customer route', async () => {
+        const mockCustomerService = sandbox.createStubInstance(CustomerService);
+        mockCustomerService.updateCustomer.resolves('foo');
+        const app = new App({
+            port: testPort,
+            controllers: [new CustomerController(mockCustomerService)],
+            middleWares: [bodyParser.json(), bodyParser.urlencoded({ extended: true })]
+        });
+        app.listen();
+        const res = await request(app.app)
+            .put('/customer/1')
+            .set('Accept', 'application/json')
+            .set('Authorization', 'bearer ' + token)
+            .send(mockCustomer);
+        expect(res.body).to.equal('foo');
+        expect(res.status).to.equal(200);
+        app.shutdown();
+    });
+
+    it('Test archive customer route', async () => {
+        const mockCustomerService = sandbox.createStubInstance(CustomerService);
+        mockCustomerService.archiveCustomer.resolves('foo');
+        const app = new App({
+            port: testPort,
+            controllers: [new CustomerController(mockCustomerService)],
+            middleWares: [bodyParser.json(), bodyParser.urlencoded({ extended: true })]
+        });
+        app.listen();
+        const res = await request(app.app)
+            .post('/customer/archive')
+            .set('Accept', 'application/json')
+            .set('Authorization', 'bearer ' + token)
+            .send(archiveCustomer);
+        expect(res.body).to.equal('foo');
+        expect(res.status).to.equal(200);
+        app.shutdown();
+    });
+
+    it('Test return archived customer route', async () => {
+        const mockCustomerService = sandbox.createStubInstance(CustomerService);
+        mockCustomerService.getAllArchivedCustomers.resolves('foo');
+        const app = new App({
+            port: testPort,
+            controllers: [new CustomerController(mockCustomerService)],
+            middleWares: [bodyParser.json(), bodyParser.urlencoded({ extended: true })]
+        });
+        app.listen();
+        const res = await request(app.app)
+            .get('/customer/archive/1')
```

Comment:
You are getting the archived customer with id of 1, but on line 166 you are mocking the get**All**ArchivedCustomers, so  you probably don't need the id of 1

Reply:
Here, the 1 is not for an id, but it's to represent the boolean true (in the backend, the 1 is evaluated as true since this test is for returning all the customers that are archived). If it was 0 instead, then this should return all non-archived customers.